File: rust/src/models/subscription_update_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::payment_methods_config::PaymentMethodsConfig;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SubscriptionUpdateRequest {
    /// If false, invoices will stay in Draft until manually reviewed and finalized.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub auto_advance_invoices: Option<bool>,

    /// Automatically try to charge the customer's configured payment method on finalize.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub charge_automatically: Option<bool>,

    /// Partial update of custom property values (merge; send a key with `null` to remove it).
    /// Validated against the tenant's `SUBSCRIPTION` property definitions. Omit to leave unchanged.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub custom_properties: Option<serde_json::Value>,

    /// Default memo for invoices
    #[serde(skip_serializing_if = "Option::is_none")]
    pub invoice_memo: Option<String>,

    /// Payment terms in days (0 = due on issue)
    #[serde(skip_serializing_if = "Option::is_none")]
    pub net_terms: Option<i32>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub payment_methods_config: Option<PaymentMethodsConfig>,

    /// Purchase order number
    #[serde(skip_serializing_if = "Option::is_none")]
    pub purchase_order: Option<String>,
}

impl SubscriptionUpdateRequest {
    pub fn new() -> Self {
        Self {
            auto_advance_invoices: None,
            charge_automatically: None,
            custom_properties: None,
            invoice_memo: None,
            net_terms: None,
            payment_methods_config: None,
            purchase_order: None,
        }
    }
}

File: typescript/src/models/subscriptionUpdateRequest.ts
// this file is @generated
import {
  type PaymentMethodsConfig,
  PaymentMethodsConfigSerializer,
} from "./paymentMethodsConfig";

export interface SubscriptionUpdateRequest {
  /** If false, invoices will stay in Draft until manually reviewed and finalized. */
  autoAdvanceInvoices?: boolean | null;

  /** Automatically try to charge the customer's configured payment method on finalize. */
  chargeAutomatically?: boolean | null;

  /**
   * Partial update of custom property values (merge; send a key with `null` to remove it).
   * Validated against the tenant's `SUBSCRIPTION` property definitions. Omit to leave unchanged.
   */
  customProperties?: any;

  /** Default memo for invoices */
  invoiceMemo?: string | null;

  /** Payment terms in days (0 = due on issue) */
  netTerms?: number | null;

  paymentMethodsConfig?: PaymentMethodsConfig | null;

  /** Purchase order number */
  purchaseOrder?: string | null;
}

export const SubscriptionUpdateRequestSerializer = {
  _fromJsonObject(object: any): SubscriptionUpdateRequest {
    return {
      autoAdvanceInvoices: object["auto_advance_invoices"],
      chargeAutomatically: object["charge_automatically"],
      customProperties: object["custom_properties"],
      invoiceMemo: object["invoice_memo"],
      netTerms: object["net_terms"],
      paymentMethodsConfig:
        object["payment_methods_config"] != null
          ? PaymentMethodsConfigSerializer._fromJsonObject(
              object["payment_methods_config"]
            )
          : undefined,
      purchaseOrder: object["purchase_order"],
    };
  },

  _toJsonObject(self: SubscriptionUpdateRequest): any {
    return {
      auto_advance_invoices: self.autoAdvanceInvoices,
      charge_automatically: self.chargeAutomatically,
      custom_properties: self.customProperties,
      invoice_memo: self.invoiceMemo,
      net_terms: self.netTerms,
      payment_methods_config:
        self.paymentMethodsConfig != null
          ? PaymentMethodsConfigSerializer._toJsonObject(self.paymentMethodsConfig)
          : undefined,
      purchase_order: self.purchaseOrder,
    };
  },
};
